pyccl/halos/concentration.py

```diff
@@ -0,0 +1,308 @@
+from .. import ccllib as lib
+from ..background import growth_factor
+from .massdef import MassDef, mass2radius_lagrangian
+from ..power import linear_matter_power, sigmaM
+import numpy as np
+
+
+class Concentration(object):
+    """ This class enables the calculation of halo concentrations.
+
+    Args:
+        mass_def (:obj:`MassDef`): a mass definition object that fixes
+            the mass definition used by this c(M)
+            parametrization.
+    """
+    name = 'default'
+
+    def __init__(self, mass_def=None):
+        if mass_def is not None:
+            if self._check_mdef(mass_def):
+                raise ValueError("c(M) relation " + self.name +
+                                 " is not compatible with mass definition" +
+                                 " Delta = %s, " % (mass_def.Delta) +
+                                 " rho = " + mass_def.rho_type)
+            self.mdef = mass_def
+        else:
+            self._default_mdef()
+        self._setup()
+
+    def _default_mdef(self):
```

Comment:
Why do we need a default? I think requiring an input from the user makes sense here.

Reply:
Hmmh, I think I disagree. There are parametrizations that are only valid for one particular mass definition (or for a couple of them). Having a default mass definition just makes your life easier, I think. I'm happy to remove this if you prefer.

Reply:
Hmmm ok. I don’t follow your argument here but I’m happy to defer to you.

Reply:
OK, thanks. For clarification, what I mean is that, for instance, the Klypin11 concentration model only works for Delta_vir masses, so forcing people to pass that mass definition every time they want to use it is kind of clunky. 
BTW, I just realized I need to add a lot of documentation to the concentration stuff. I'll do that now.

Reply:
Alright! Sounds good!